roles/download/defaults/main.yml

```diff
@@ -60,7 +60,7 @@ downloads:
     url: "{{ calico_cni_download_url }}"
     owner: "root"
     mode: "0755"
-    enabled: "{{ kube_network_plugin == 'calico' or kube_network_plugin == 'canal' }}"
+    enabled: "{{ kube_network_plugin == 'calico' and not use_calico_cni_container|default(false) }}"
```

Comment:
I'm sure. Do you have a use case where download is needed and calico/cni container won't work?